Feature/lint and format

AKMaily opened the pull request:
This PR is currently only in draft so i can mention part of the code in another PR. 

It does not need to be reviewed at the moment. 



## Summary

**Brief description about the content of your PR and its improvements for the project.**

1. What problem does this PR solve? Which concept, bug, or requirement does it address?

2. Please add context to the improvements: reference an issue ("Resolves #<your issue id here>"), a user story, or explain the improvements in documentation, code, or UX.

## Design Decisions

**Describe the way your implementation works or what design decisions you made if applicable.**

Which are the main files and concepts you changed or introduced?

## Testing

**Briefly explain how it can be tested—either with commands, manual steps, or test files.**

1. Provide a description of the expected behavior.

2. Describe how you tested the changes.

3. Is there still unexpected behaviour which needs to be addressed in the future?

## Checklist

Make sure you

- [ ] have read the [contribution guidelines](../CONTRIBUTION.md)
- [ ] have added necessary unit/e2e tests if necessary.
- [ ] have added documentation if necessary.

tabeatheunicorn commented on CHANGELOG.md:
Please undo the whitespace changes and only commit your adaptions. Otherwise this leads to noise imho

tabeatheunicorn replied:
Update: is okay as this is part of the formatting commit

tabeatheunicorn commented on README.md:
```diff
@@ -22,7 +22,7 @@ We want to make this better!
 We are very happy about contributions. If you want to contribute just follow the steps in the [CONTRIBUTIONS.md](CONTRIBUTION.md). We will review your contribution as fast as possible.
 
-## How to get the Code Documentation : 
+## How to get the Code Documentation :
```
Please don't commit whitespace changes

tabeatheunicorn reviewed (changes requested):
This PR has two tasks:
First the setup of formatting and linting and second the fixes resulting from issues of formatting and linting.
The general setup for formatting and linting should be discussed and imho that will be easier when there are two PRs: One introducing the setup and one fixing the changes or the history is condensed in two commits - setup and fixes

AKMaily commented:
> This PR has two tasks: First the setup of formatting and linting and second the fixes resulting from issues of formatting and linting. The general setup for formatting and linting should be discussed and imho that will be easier when there are two PRs: One introducing the setup and one fixing the changes or the history is condensed in two commits - setup and fixes

Went with the first option and reverted the format and lint changes .

tabeatheunicorn reviewed (changes requested):
This is missing a verification check in the ci pipeline so that PR authors get notified about wrong formatting

C0D3-M4513R reviewed (approved):
Looks good to me.

Some of the diffs were hard to read, because of large changed sections, but overall I'm pretty confident, that the behavior should be the same everywhere.

Overall nice linter/prettyfier changes though.
CI changes also look clean

AKMaily commented:
> Looks good to me.
> 
> Some of the diffs were hard to read, because of large changed sections, but overall I'm pretty confident, that the behavior should be the same everywhere.
> 
> Overall nice linter/prettyfier changes though. CI changes also look clean
> 
> The only concern to me is what the process for fixing up pr's would look like, as there are bound to be a lot of merge conflicts with pr's (and also other outside branches, which are not being pr'd).

Thank you for the feedback. 
I think all the format changes should be registered by git only as format changes and not lead to merge conflicts. 
The only changes i think that will be registered are the manual linter changes, those were 17. I think it depends on the PR branch how those are registered but it should be handable to merge those. @tabeatheunicorn what is your insight on this ?

AKMaily commented on CHANGELOG.md:
```diff
@@ -46,11 +48,8 @@ Formatting needs to be similar to the dataformat of the old OmnAIView data expor
 
 - Deletion of deprecated Angular 18 patterns (#18)
-- Deletion of duplicated AsyncAPI description for OmnAIBackend 
+- Deletion of duplicated AsyncAPI description for OmnAIBackend
```
I think this uses the same wording as above but as the Changelog seems to switch very often in the verb tense . Would change that in a different PR as this will not be touched after reverting the formatter and linter.

AKMaily commented on .github/workflows/main.yml:
```diff
@@ -37,6 +37,10 @@ jobs:
+      - name: Run ESLint Check
+        run: npm run lint:check
+      - name: Run Prettier check
+        run: npm run format:check
```
Done

AKMaily commented:
> > changes i think that will be registered are the manual linter changes, those were 17. I think it depends on the PR branch how those are registered but it should be handable to merge those. @tabeatheunicorn what is your insight on this ?
> 
> I think that we should have two separate PRs as already suggested [#75 (review)](https://github.com/omnai-project/OmnAIView/pull/75#pullrequestreview-2942761451): This PR should only add the CI Pipeline and setups for linter and formatter. That way, any developer can fix issues on their own or cherry-pick changes from the second PR. Whitespace changes, line breaks etc. will lead to very odd merge conflicts, hence I would strongly favor separating the fixes and the setup

Changed this and added the new PR in #103 . It is now expected for this code that the style CI job will not succeed.

C0D3-M4513R commented on .github/workflows/main.yml:
```diff
@@ -50,3 +50,10 @@ jobs:
             electron\out\make\squirrel.windows\x64\RELEASES
             electron\out\make\squirrel.windows\x64\*.nupkg
             electron\out\make\squirrel.windows\x64\*.exe
+  CheckStyle: 
+    runs-on: windows-latest
+    steps: 
+      - name: Run ESLint Check
+        run: npm run lint:check
+      - name: Run Prettier check
+        run: npm run format:check
```
This part will fail, since separate github action checks are built on separate runners.
Nothing is carried over.

This means, that you need to duplicate the setup bits (or somehow put them in a dedicated file, which is pulled into each action)
```suggestion
  CheckStyle: 
    runs-on: windows-latest
    steps: 
      - name: Repository-Checkout
        uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: ${{ env.NODE_VERSION }}
          cache: 'npm'
          cache-dependency-path: |
            package-lock.json
      - name: Cache node_modules
        id: cache-node_modules
        uses: actions/cache@v4
        with:
          path: node_modules
          key: node_modules-${{ runner.os }}-${{runner.arch}}-node_${{ env.NODE_VERSION }}-${{ hashFiles('package-lock.json') }}
      - name: Install NPM-Dependencies
        if: ${{steps.cache-node_modules.outputs.cache-hit != 'true'}}
        run: npm ci --no-audit --no-fund
      - name: Run ESLint Check
        run: npm run lint:check
      - name: Run Prettier check
        run: npm run format:check
```

C0D3-M4513R reviewed (changes requested):
Github Actions status-check will never pass

C0D3-M4513R reviewed (changes requested):
Github Actions status-check will never pass

C0D3-M4513R reviewed (approved):
As I have understood the intent of this pr is to only add the linting(and #103 should then fix the lints), so the linting ci failing is expected.

With that in mind, this looks good.
